### DISCUSSION TITLE
Text Format Flags on DecoratorBlockNode (or DecoratorNode)

### DISCUSSION THREAD

**Discussion by srinivasskeme** [facebook/lexical#4971] [General]
I have a TagNode with the following requirements. 

- Once a tag node is inserted, its text cannot be edited. 
- I would also like to have a outline on the node when the node is selected. (i.e., I don't want cursor inside the TagNode). 

Because of these two requirements, `DecoratorBlockNode` seems like the correct choice.

However, I also need to have text format features (like bold, italic etc.) on the tag node. I have the following solutions in mind but I am unsure if they work. 

Solution 1:

1. Add extra fields to `TagNode` (`__textFormatFlags`, `__textStyle`)
2. register listener to `FORMAT_TEXT_COMMAND` and ensure both `TextNode` and `TagNode` are updated (bold, italic etc.)
3. implement custom variant of `$patchStyleText` to handle (`font-size`, `font-family `etc.) for `TextNode` and `TagNode`

Solution 2:
1. Add a nested LexicalEditor inside `TagNode` in disabled mode
2. use similar listeners approach as `Solution 1` to update text nodes within nested editor

What is the better solution and are there any gotchas that I am overlooking?

**Comment by liuweiGL** [comment]
It would be nice to combine TextNode with DecoratorNode, is that possible?

**Reply by srinivasskeme** [reply]
Not to my knowledge. Neither inherits from the other node.
Additionally, lexical is aware of these nodes. The functionality to add text formatting uses `$isTextNode`. Similarly, the decorate function is only called if `$isDecoratorNode` is called.

Note: I don't have access to my system now. So the function names might be slightly different.
